### ISSUE TITLE
Worker manager provisioner DDoSing web service

### ISSUE THREAD

**Issue by lotas** [taskcluster/taskcluster#5441]
Incident: 2022-05-11 Worker Manager

Api calls started to get slower: from 100s of ms to 1s and up

`17:45:36 CEST` (2022-05-11T15:45:36.543810767Z)  Api calls started to get slower: from 100s of ms to 1s and up

`17:47:00 CEST` those calls were taking up to 1min
`17:47:10 CEST` + api calls begin to be terminated by nginx after 60s  

Overall count of `registerWorker` calls rose from 30 calls/minute to 1k-2k per minute

Investigating provisioning:

`17:44:26 CEST`  worker manager requests 167 `gecko-t/t-linux-large` instances
`17:44:36 CEST`  worker manager requests 167 `gecko-t/t-linux-large` instances
`17:44:44 CEST`  worker manager requests 167 `gecko-t/t-linux-large` instances
`17:44:52 CEST`  worker manager requests 167 `gecko-t/t-linux-large` instances
`17:45:00 CEST`  worker manager requests 167 `gecko-t/t-linux-large` instances
`17:45:12 CEST`  worker manager requests 167 `gecko-t/t-linux-large` instances
`17:45:20 CEST`  worker manager requests 167 `gecko-t/t-linux-large` instances
`17:45:32 CEST`  worker manager requests 167 `gecko-t/t-linux-large` instances

So in just 1 minute there was `+1336` instances requested
Minute later they were booted and started to send `registerWorker` calls to the api

`worker-manager-web` deployment had only 1 working pod, so it failed to serve all incoming requests and resulted in all api calls being timed out (also UI calls)

Incident was resolved around `20:25:00 CEST` by scaling worker-manager-web to 9 pods

Root cause of the issue:
1. worker manager provisioner requests instances and discards the ones that were already spinning up (thus multiple requests of 167 workers)
2. web service was not scaled enough to handle increased load

**Comment by lotas** [comment]
Sentry reported around the same time:

```
Error                        
Provisioner.provision(provisioner)                                              
provision loop interference                                                                         
Seen 75 times.              
Last seen: May 11, 2022, 6:32 p.m. UTC             in                          operations / taskcluster-firefox-ci
```

which might be related to a fix from #5003

**Comment by lotas** [comment]
It looks like there is nothing bad in the work of the provisioner, at the beginning of the loop it had those:

```
    "Fields": {
      "minCapacity": 0,
      "maxCapacity": 2500,
      "desiredCapacity": 1897,
      "workerPoolId": "gecko-t/t-linux-large",
      "scalingRatio": 1,
      "requestedCapacity": 0,
      "existingCapacity": 230,
      "pendingTasks": 1667
    },
}
```

So from existing capacity of `230` and pending tasks `1667` it had to request additional workers.
